documentation/rules/README.md

```diff
@@ -0,0 +1,26 @@
+
+## Rules

```

Comment:
Should we change the link from https://github.com/nsubstitute/NSubstitute.Analyzers/blob/dev/documentation/README.md to point us to this rule summary instead of rules directory?

Reply:
This is another question I wasn't sure about. 😕 

It probably looks better and is most useful to have the table included directly in the main readme. Maybe I can do this as a separate task?

Reply:
For now I will update the link to point to the rules summary. 👍